website/docs/language/checks/index.mdx

```diff
@@ -0,0 +1,89 @@
+---
+page_title: Checks - Configuration Language
+description: >-
+  Check customized infrastructure requirements to provide ongoing and continuous verification.
+---
+
+# Checks
+
+-> **Note:** Check blocks are only available in Terraform v1.5.0 and later.
+
+The `check` block can validate your infrastructure outside the usual resource lifecycle. Check blocks address a gap between post-apply and functional validation of infrastructure. Check blocks allow you to define [custom conditions](/terraform/language/expressions/custom-conditions) that execute on every Terraform plan or apply operation without affecting the overall status of an operation. Check blocks execute as the last step of a plan or apply, after Terraform has planned or provisioned your infrastructure.
```

Comment:
> Check blocks allow you to define [custom conditions](/terraform/language/expressions/custom-conditions) that execute on every Terraform plan or apply operation without affecting the overall status of an operation.

Forgive me, I'm still trying to work out the similarities and differences between checks and custom conditions. My understanding was that Custom Conditions™ included preconditions, postconditions, and variable validation, and we wanted to present check blocks as something distinct. 

If that's true, the language above is confusing to me. On one hand I can see how the functionality of check blocks could be described using the words "custom condition", but if checks are distinct enough to warrant their own section in the docs then I might suggest not linking out to the other feature here.

_On the other hand_, I am really wondering if it makes sense that checks land under the umbrella of Custom Conditions™, since they kind of are that. Should we consider moving this content under the Custom Condition Checks page? Does anyone else agree that might be less confusing? Am I missing any reasons we couldn't do that? Is it too late?

Reply:
Please disregard the message above! It was premature. I read through the rest of the PR and the way everything fits together is making more sense to me now.